### PR TITLE
feat(Exception Handler): Failsafe Exception Handler, updates QueueableProcess

### PR DESCRIPTION
**Pull request by codefriar**
This commit introduces a Failsafe Exception handler with Callable interface. This allows other bits of Apex kit to safely handle exceptions via logging, *if* the failsafeExceptionHandler is part of the org. if it's not... well the exceptions are silently handled.

### CODE REVIEW

**Comment by codefriar** [comment]
@jamessimone i've exposed the full history of both the queueable context and the finalizer context. Also exposed a method to handle exceptions on a per-class basis

**Comment by dschach** [comment]
@codefriar Multiple PMD violations - I will attempt to fix them

**Comment by dschach** [comment]
@codefriar The PMD ruleset for deployment requires ApexDoc for all public methods. Do you want to update the violations in this PR and open a separate PR to fix for files not included in this PR?

* QueueableProcess.cls
* QueueableProcessDataProvider.cls
* QueueableProcessTests.cls
* ExampleQueueableProcessSteps

**Review by dschach — CHANGES_REQUESTED** [review]
Update ApexDocs as needed, please

File: force-app/main/default/classes/QueueableProcess/EnqueueNextQueueableProcessStep.cls [hunk 1]
```diff
@@ -46,18 +67,43 @@ public class EnqueueNextQueueableProcessStep implements System.Finalizer {
...
 			}
 		}
 	}
+
+	private void handleException(FinalizerContext context) {
+		try {
+			String className = this.dataProvider.getClassName(
+				context.getAsyncApexJobId()
+			);
+			QueueableProcess parentQueueable = (QueueableProcess) Type.forName(
```
Comment (dschach) [inline force-app/main/default/classes/QueueableProcess/EnqueueNextQueueableProcessStep.cls:86]: @codefriar Looks like test is failing here

**Review by dschach — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/queueable process/EnqueueNextQueueableProcessStep.cls [hunk 2]
```diff
@@ -37,23 +57,59 @@ public class EnqueueNextQueueableProcessStep implements System.Finalizer {
...
 			}
 		}
 	}
+
+	private void handleException(FinalizerContext context) {
+		try {
+			String className = this.dataProvider.getClassName(
+				context.getAsyncApexJobId()
+			);
+			QueueableProcess parentQueueable = (QueueableProcess) Type.forName(
```
Comment (dschach) [inline force-app/main/default/classes/queueable process/EnqueueNextQueueableProcessStep.cls:92]: @codefriar This is where the test has an error.

```
System.NullPointerException: Attempt to de-reference a null object                                                  
External entry point                                                                                                
Class.EnqueueNextQueueableProcessStep.handleException: line 92, column 1                                            
Class.EnqueueNextQueueableProcessStep.execute: line 82, column 1                                                    
Class.QueueableProcessTests.testFinalizerHandlesUnhandledExceptionPositiveFailsafeHandlerCalled: line 74, column 1  35

**Review by dschach — CHANGES_REQUESTED** [review]

**Review by dschach — APPROVED** [review]

**Pull request merged**